### PR TITLE
Add custom message to terminator

### LINKED ISSUE #17000: Add optional message to terminator UO
## Reason
I have a use case where several terminators are checking simulation conditions and in some cases cause a successful termination of a completed run (as soon as the given condition is hit), but in other cases I want the simulation to fail with an error (or warning message).

## Design
Add optional message, and - if the message is provides - an error level (INFO, WARNING, ERROR) for the message (defaulting to INFO).

## Impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #17000

```diff
#Time Step 0, time = 0
#
#Postprocessor Values:
#+----------------+----------------+
#| time           | time           |
#+----------------+----------------+
#|   0.000000e+00 |   0.000000e+00 |
#+----------------+----------------+
#
#
#Time Step 1, time = 1, dt = 1
# 0 Nonlinear |R| = 1.954340e+00
# 1 Nonlinear |R| = 7.101500e-06
# 2 Nonlinear |R| = 4.846944e-11
+ Solve Converged!
#arnold1 is marking the current solve step as failed.
#
#
@@*** Info ***@@
@@This is an info@@
#Aborting as solve did not converge
#
#Time Step 1, time = 0.5, dt = 0.5
# 0 Nonlinear |R| = 1.563472e+00
# 1 Nonlinear |R| = 4.700734e-06
# 2 Nonlinear |R| = 2.908343e-11
+ Solve Converged!
#arnold2 is marking the current solve step as failed.
#
#
!*** Warning ***
!The following warning occurred in the object "arnold2", of type "Terminator".
!
!This is a warning!
#Aborting as solve did not converge
#
#Time Step 1, time = 0.25, dt = 0.25
# 0 Nonlinear |R| = 7.817360e-01
# 1 Nonlinear |R| = 6.333460e-08
# 2 Nonlinear |R| = 2.079589e-14
+ Solve Converged!
#
#
-*** ERROR ***
-The following error occurred in the object "arnold3", of type "Terminator".
-
-This is an error!
#
#Stack frames: 20
```

### CODE REVIEW

**Comment by dschwen** [comment]
Example use case is checking that a solute concentration does not exceed the valid range for a given thermodynamic model (and if it does, error out with the appropriate message), but terminate the simulation successfully once multiple precipitates have coarsened into a single precipitate.

**Review by loganharbour — DISMISSED** [review]
Screenshots of console output 😢

File: test/tests/userobjects/Terminator/tests [hunk 1]
```diff
@@ -18,4 +18,31 @@
     issues = '#16452'
     requirement = "The system shall be able to terminate abort the current step in a solve based on post-processing calculations performed within the simulation to cut the timestep."
   [../]
+  [./terminator_message_1]
+    type = 'RunException'
+    allow_warnings = true
+    expect_err = "This is an error!"
+    input = 'terminator_message.i'
+    group = 'requirements'
```
Comment (loganharbour) [inline test/tests/userobjects/Terminator/tests:26]: Why do we have this? remnants of older SQA?

**Comment by dschwen** [comment]
> Screenshots of console output 😢

To show the colorful output of the different available error levels

File: test/tests/userobjects/Terminator/tests (hunk 1 above)
Comment (dschwen) [reply test/tests/userobjects/Terminator/tests:26]: No idea. @aeslaughter ? There are about 33 of those in framework tests.

Comment (aeslaughter) [reply test/tests/userobjects/Terminator/tests:26]: Yes, this is old and can be removed.

Comment (dschwen) [reply test/tests/userobjects/Terminator/tests:26]: Ok, I'll add a commit for that

**Review by loganharbour — APPROVED** [review]
I take back my previous comment.

I appreciate the potato quality image of console output.

**Pull request merged**